notification-channel-config-service-impl/src/main/java/org/hypertrace/notification/config/service/NotificationChannelConfigServiceRequestValidator.java

```diff
@@ -0,0 +1,51 @@
+package org.hypertrace.notification.config.service;
+
+import static org.hypertrace.config.validation.GrpcValidatorUtils.validateNonDefaultPresenceOrThrow;
+import static org.hypertrace.config.validation.GrpcValidatorUtils.validateRequestContextOrThrow;
+
+import com.google.common.base.Preconditions;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.notification.config.service.v1.CreateNotificationChannelRequest;
+import org.hypertrace.notification.config.service.v1.DeleteNotificationChannelRequest;
+import org.hypertrace.notification.config.service.v1.GetAllNotificationChannelsRequest;
+import org.hypertrace.notification.config.service.v1.NotificationChannel;
+import org.hypertrace.notification.config.service.v1.NotificationChannelMutableData;
+import org.hypertrace.notification.config.service.v1.UpdateNotificationChannelRequest;
+
+public class NotificationChannelConfigServiceRequestValidator {
+
+  public void validateCreateNotificationChannelRequest(
+      RequestContext requestContext, CreateNotificationChannelRequest request) {
+    validateRequestContextOrThrow(requestContext);
+    validateNotificationChannelMutableData(request.getNotificationChannelMutableData());
+  }
+
+  public void validateUpdateNotificationChannelRequest(
+      RequestContext requestContext, UpdateNotificationChannelRequest request) {
+    validateRequestContextOrThrow(requestContext);
+    validateNonDefaultPresenceOrThrow(
+        request.getNotificationChannel(), NotificationChannel.ID_FIELD_NUMBER);
+    validateNotificationChannelMutableData(
+        request.getNotificationChannel().getNotificationChannelMutableData());
+  }
+
+  private void validateNotificationChannelMutableData(NotificationChannelMutableData data) {
+    validateNonDefaultPresenceOrThrow(
+        data, NotificationChannelMutableData.CHANNEL_NAME_FIELD_NUMBER);
+    Preconditions.checkArgument(
+        data.getEmailChannelConfigCount() != 0 || data.getWebhookChannelConfigCount() != 0,
```

Comment:
The logic makes sense, but can we throw this as an invalid arg status exception.

Reply:
Although, should we be checking each email/webhook config for validity, too?

Reply:
Ideally yes, but previously there wasn't any check. 

Basically we need to validate the url and email address. Probably use some util to do that. I'll maybe do it later on?

Reply:
Sure